File: hassan.js
const { Client, GatewayIntentBits } = require('discord.js');
const client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.GuildMessageReactions,
    GatewayIntentBits.GuildIntegrations
  ],
});

const fetch = require('node-fetch');

const jokes = [
  'Why did the tomato turn red? Because it saw the salad dressing!', 
  'Why was the math book sad? Because it had too many problems.', 
  'What do you call fake spaghetti? An impasta.', 
  'What do you call a bear with no teeth? A gummy bear.', 
  'Why don’t scientists trust atoms? Because they make up everything.'
];

const facts = [
  'Pakistan is the world’s 6th most populous country.', 
  'Pakistan’s official name is “Islamic Republic of Pakistan”', 
  'Pakistan was first created on August 14, 1947.', 
  'Pakistan is the only Muslim nuclear power country.', 
  'Shahid Afridi has hit 476 sixes in his international cricket career.'
];

client.on('ready', () => {
  console.log("I'm in");
  console.log("Hassan Bot");
});

client.on('interactionCreate', async interaction => {
  if (!interaction.isCommand()) {
    return;
  }

  const { commandName } = interaction;

  if (commandName === 'info') {
    await interaction.reply('Hi! My name is Hassan Bot. Named after Mohammed "Hassan" Khan, a famed member of the community and a dear friend of the creator. I was created by Malek Kamal, who worked day and night to make me work the way I do! I am currently in development version 1.0. Thanks for listening to me!');
  } else if (commandName === 'joke') {
    const randomJoke = jokes[Math.floor(Math.random() * jokes.length)];
    await interaction.reply(randomJoke);
  } else if (commandName === 'fact') {
    const randomFact = facts[Math.floor(Math.random() * facts.length)];
    await interaction.reply(randomFact);
  } else if (commandName === 'math') {
    const num1 = parseInt(interaction.options.getString('num1'));
    const num2 = parseInt(interaction.options.getString('num2'));
    const operator = interaction.options.getString('operator');

    let result;

    switch (operator) {
      case '+':
        result = num1 + num2;
        break;
      case '-':
        result = num1 - num2;
        break;
      case '*':
        result = num1 * num2;
        break;
      case '/':
        result = num1 / num2;
        break;
      default:
        result = 'Invalid operator';
        break;
    }

    await interaction.reply(result.toString());
  }
});

client.on('messageCreate', async message => {
  const msg = message.content.toLowerCase();

  if ((msg.includes('hassan')) && (message.author.id !== client.user.id)) {
    await message.reply('Who summoned me?');
  }

  if ((msg.includes('shut up')) && (message.author.id !== client.user.id)) {
    await message.reply('I am not supposed to respond to that.');
  }
});

client.login(process.env.DISCORD_BOT_SECRET);

const keepAlive = () => {
setInterval(() => {
    const url = 'https://discord.com/api/v8/users/@me/channels';
    fetch(url, {
      method: 'GET',
      headers: {
      Authorization: `Bot ${process.env.DISCORD_BOT_SECRET}`,
      },
    })
    .then(res => {
      console.log('pinged');
    });
  })}

File: hassan.py
import os
import random
import discord
from keep_alive import keep_alive

intents = discord.Intents.default()
intents.message_content = True

class HassanBot(discord.Client):
    def __init__(self, **options):
        super().__init__(**options)

        # List of jokes
        self.jokes = [
            "Why did the scarecrow win an award? Because he was outstanding in his field.",
            "What do you call a fish with no eyes? Fsh!",
            "What do you call a cow with no legs? Ground beef.",
            "Why did the scarecrow win an award? Because he was outstanding in his field.",
            "What do you call a fish with no eyes? Fsh!",
            "What do you call a cow with no legs? Ground beef.",
        ]

        # List of facts about Pakistan
        self.facts = [
            "Pakistan is the fifth-most populous country in the world.",
            "Pakistan is the world's largest Muslim-majority country.",
            "Pakistan is home to the world's second-highest mountain, K2.",
            "Pakistan is the world's seventh-largest nuclear power.",
            "Pakistan is a member of the United Nations, the Organisation of Islamic Cooperation, and the Commonwealth of Nations.",
        ]

        # List of unallowed words
        self.unallowed_words = ["shit", "fuck", "bitch", "nigger", "nigga", "kys", "dick", "pussy", "asshole", "bastard", "damn", "kinky", "suck", "kys" "self harm" "self-harm"]

    async def on_ready(self):
        print('Hassan Bot is ready!')

    async def on_message(self, message):
        # Respond to the /info command
        if message.content.startswith('/info'):
            await message.channel.send('Hi! My name is Hassan Bot. Named after Mohammed "Hassan" Khan, a famed member of the community and a dear friend of the creator. I was created by Malek Kamal in 2023, who worked day and night to make me work the way I do! I am currently in development version 1.0. Thanks for asking!')

        # Detect unallowed words
        unallowed_words = self.detect_unallowed_words(message.content)

        # Ping the user and say "Haram you little gremlin" if any unallowed words are found
        if unallowed_words:
            await message.channel.send(f'<@{message.author.id}>, Haram! Same to you, you fat loner.')

        # Respond to the /joke command
        elif message.content.startswith('/joke'):
            await message.channel.send(random.choice(self.jokes))

        # Respond to the /fact command
        elif message.content.startswith('/fact'):
            await message.channel.send(random.choice(self.facts))

        # Respond to the /math command
        elif message.content.startswith('/math'):
            try:
                result = eval(message.content[6:])
                await message.channel.send(str(result))
            except Exception as e:
                await message.channel.send(f'Error: {e}')

        # Respond to messages that mention the word "Hassan"
        elif 'Hassan' in message.content:
            await message.channel.send('Who has summoned me?')

        # Respond to messages that say "shut up"
        elif message.content == 'shut up':
            await message.channel.send('Not in my good Muslim house you shall say that :-(')

    def detect_unallowed_words(self, message):
        unallowed_words = []
        for word in message.split():
            if word in self.unallowed_words:
                unallowed_words.append(word)

        return unallowed_words

if __name__ == '__main__':
    HassanBot = HassanBot(intents=intents)
    HassanBot.run(os.getenv('TOKEN'))

async def on_member_join(self, member):
    # Get the channel object by name (replace "general" with the channel you want)
    channel = discord.utils.get(member.guild.channels, name='announcements')

    # Ping the user and say "Welcome to the server!"
    await channel.send(f'Welcome to the server, <@{member.id}>! I hope you find our server welcoming :)')

if __name__ == '__main__':
    HassanBot = HassanBot(intents=discord.Intents.default())
    keep_alive()
HassanBot.run(os.getenv('TOKEN'))
